lib/contextual_logger/logger_with_context.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module ContextualLogger
+  # A logger that deep_merges additional context and then delegates to the given logger.
+  # Keeps it own log level that may be set independently of the logger it delegates to (the latter's log level is ignored).
+  class LoggerWithContext
+    include LoggerMixin
+
+    attr_accessor :level
+    attr_reader :context
+
+    def initialize(logger, context, level: nil)
+      @logger = logger
+      @level = level || logger.level
```

Comment:
Hmm, if you don't set a `level` here, the `LoggerWithContext` will snap to the `level` in the passed-in logger. But it copies it here, meaning it won't track changes made below. Will this be surprising? The alternative that just occurred to me is that I could leave `nil` to mean "I always want follow the lower logger's level".

Reply:
I think it would be better to not copy the value, and instead leave it as fully inherited from the parent logger.

Reply:
Agreed. I'll change it.

Reply:
Fixed.